src/util/CrosisError.ts

```diff
@@ -0,0 +1,22 @@
+export default class CrosisError extends Error {
+  /** extras - a map of event data in a format appropriate to send to an error logging service */
+  public readonly extras: Record<string, unknown> | null;
+  /** tags - a map of tags to their values in a format appropriate to send to an error logging service */
+  public readonly tags: Record<string, string> | null;
+
+  constructor(
+    message: string,
+
+    extras: Record<string, unknown> | null = null,
+    tags: Record<string, string> | null = null,
+  ) {
+    super(message);
+
+    this.extras = extras;
+    this.tags = tags;
```

Comment:
```suggestion
    super(message);

    this.name = 'CrosisError';
    this.extras = extras;
    this.tags = tags;
    
    if ('captureStackTrace' in Error) {
      Error.captureStackTrace(this, CrosisError);
    } else {
      this.stack = Object.defineProperty(this, 'stack', {
        enumerable: false,
        value: Error(message).stack,
        writable: true,
        configurable: true,
      });
    }
```

Most of the time the stack stuff is unnecessary, but iirc i had an issue with stack traces in the past when i didn't do that.

Reply:
More importantly is that we set `name`

Reply:
Yeah, maybe the stack stuff is not needed in this context